services/wireguard/resources/allocator_windows.go

```diff
@@ -61,7 +60,7 @@ func (a *Allocator) AllocateIPNet() (net.IPNet, error) {
 	a.mu.Lock()
 	defer a.mu.Unlock()
 
-	for i := 1; i < MaxResources; i++ {
+	for i := 2; i < a.maxConnections+2; i++ {
```

Comment:
could you perhaps add a comment explaining this loop? why is i 2, why a.maxConnections+2? perhaps adding new variables could also help.